src/components/carousel/carousel.js

```diff
@@ -424,35 +503,52 @@ export default {
       })
     )
 
+    const on = {
+      mouseenter: this.pause,
+      mouseleave: this.restart,
+      // focusin: this.pause,
+      // focusout: this.restart,
+      keydown: (evt) => {
+        if (/input|textarea/i.test(evt.target.tagName)) {
+          return
+        }
+        const keyCode = evt.keyCode
+        if (keyCode === KeyCodes.LEFT || keyCode === KeyCodes.RIGHT) {
+          evt.preventDefault()
+          evt.stopPropagation()
+          this[keyCode === KeyCodes.LEFT ? 'prev' : 'next']()
+        }
+      }
+    }
+    // Touch support event handlers for environment
+    if (!this.noTouch && hasTouchSupport) {
+      // Attach appropriate listeners
+      if (hasPointerEvent) {
+        on.pointerdown = this.touchStart
+        on.pointerup = this.touchEnd
+      } else {
+        on.touchstart = this.touchStart
+        on.touchmove = this.touchMove
+        on.touchend = this.touchEnd
+      }
+    }
+
     // Return the carousel
     return h(
       'div',
       {
-        class: [
-          'carousel',
-          'slide',
-          { 'carousel-fade': this.fade }
-        ],
+        staticClass: 'carousel slide',
```

Comment:
I would suggest to add the `slide` class to the `class` property.

Reply:
But the class "slide" is always on the carousel, so it would be considered static (i.e. not a dynamic class).

Reply:
Just added in an prop to disable slide/fade animation, so `slide` is now a dynamic class.